### ISSUE TITLE
Temporal conditions not working

### ISSUE THREAD

**Issue by particulate-matter** [osmandapp/OsmAnd#20350]
### Description

As far as I know, OsmAnd is supposed to understand and report the maxspeed:conditional time-based speed limits. However, it isn't working for me. I have "Consider temporary limitations" turned on. 

### Steps to reproduce

Open OsmAnd, switch to a driving profile.

Menu -> Configure Screen -> driving profile -> Speedometer, turn it on. Settings -> Speed limit warning, set it to Always.

Menu -> Navigation -> Settings, ensure "Consider temporary limitations" is on.

Drive on this portion of road between 7am and 8pm (Australia/Melbourne time) any day of the year https://www.openstreetmap.org/way/711800833#map=19/-37.85033/145.09494

### Actual result

Speed limit is reported as 60 km/h

### Expected result

Speed limit is reported as 40 km/h. But if you drive there overnight after 8pm or before 7am, speed limit should be reported as 60 km/h

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version: + 4.7.17 released 2024-05-07
Android/iOS version: 14 2024-07-05
Device model: Pixel 7
Crash-Logs: ?
```

Maps for Australia Victoria dated 2024-07-01, live updates turned on Updated Today - 17:05.

**Comment by particulate-matter** [comment]
I also note that another user has reported something similar on 2024-03-19 on the t.me/OsmAndMaps telegram channel, but the conversation is abandoned before it reaches a conclusion.

**Comment by DmitryAlexei** [comment]
https://osmand.net/map/navigate/?start=-37.849906,145.095644&end=-37.850478,145.093767&profile=car#19/-37.85046/145.09457

**Comment by DmitryAlexei** [comment]
Issue (if persisted) was fixed in OsmAnd 4.8.3, released: 2024-07-18 (available for beta-testing)
- I configured date and time to match 
maxspeed:conditional | 40 @ (07:00-20:00)
for 19:53 (phone time) speed limit is 40 km/h and for 21:28 speed limit is 60 km/h
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/6dd8b0a7-3dcf-4c71-941d-40ccb09c14cc"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/469e778c-2346-4174-a486-d8576c1e4c77"  width="300" height="auto"/> </td>
  </tr>
</table>

**Comment by particulate-matter** [comment]
I never got offered the beta for 4.8.3, it went straight from 4.8.2 to 4.8.4 in my store.

After I upgraded to 4.8.4, it wasn't fixed. However, I uninstalled it and reinstalled it, and after that it worked. I suppose there was some
